### ISSUE TITLE
Support of Keyboard-only navigation in Communication Sites and limited accessibility

### ISSUE THREAD

**Issue by StfBauer** [SharePoint/sp-dev-docs#1678]
#### Category
- [x] Question

#### Expected or Desired Behavior
All elements of a modern communication site should be accessible through keyboard navigation.

#### Observed Behavior
* Unaccessible navigation on communication site because the first selected element is always the search
* Edit Button is not accessible
* Follow and Share Site is not accessible
* It seems like the accessibility is only meant to consume content but not edit or modify content
* Not related to Communication Site but it seems that the suite bar, in general, is not intended to be used through keyboard-only navigation.
* Tabbing through the browser work sometimes in a special way: Sometimes it works on various sections by pressing the arrow keys. This is pretty uncommon and the description is only available on screen readers. The problem with that not all keyboard only use might use screen readers at all.

**Comment by lucaband** [comment]
hey Stefan, 
As we are unable to reproduce this in house, let me ask you a couple of additional questions: 

1. What browser did this repro on? We optimize for Narrator and Edge.

2. Would be possible for you to attach a video and/or share a site that where the issue is happening so that we can take a closer look at for this? 

 3. What are the actions you are trying to navigate? Are those just tabbing through to all the controls? That is not correct keyboarding behavior. Tabbing should be between controls and we use arrow keys to navigate within controls. This allows users to be able to get to the main areas of the app faster.

5. We also have well-defined regions and landmarks on the communication sites: you should be able to use these with a screenreader to navigate faster as well

Thanks!

**Comment by StfBauer** [comment]
Hey Luca,

I recorded a video for you that shows the [NVDA](https://youtu.be/u81ytyrb-S0) one of the two most common screen readers I found in the following [statistic](https://webaim.org/projects/screenreadersurvey7/)
Sadly I wasn't able to record the keystrokes.

As you see in the video sometimes I had to scroll manually to browse the page. 

Meanwhile based on your feedback I also tested it with the default Narrator and Edge and this works way better, But according to the survey is only used by 0.3% of users.
I'm sure you have better metrics than I get publicly.

In addition, I found the following interesting link on "[Understanding screen reader interaction modes](https://tink.uk/understanding-screen-reader-interaction-modes/)" which might explain some of the behaviours I experienced.

I might have been completely wrong because my skill level of web accessibility is meanwhile a bit rusty. I currently try to get back on track in this area.

Thank you very much for your feedback!!!

**Comment by wobba** [comment]
@lucaband Good example. All result items on SharePoint Home when doing search has `tabindex="-1"` effectivly removing them from tab order. Change to `tabindex="0"` and accessibility is covered :)

**Comment by StfBauer** [comment]
No feedback received yet.
